### PR TITLE
fix(Area): fixed a bug where className was not assigned to areaDot

### PR DESCRIPTION
**Pull request by 108yen**
## Description

Fixed a bug where classname was not assigned when passed to areachart dot as follows.
```tsx
          <Area
            type="monotone"
            dot={{ className： 'custom-dot-className' }}
            dataKey="uv"
            stroke="#8884d8"
            fill="#8884d8"
          />
```
## Related Issue

#4290 

## Motivation and Context

Required if you want to assign a className to dot in areachart.

## How Has This Been Tested?

Checked in storybook and passed existing tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by ckifer — CHANGES_REQUESTED** [review]
Thanks for the contribution! Just one thing

File: src/cartesian/Area.tsx [hunk 1]
```diff
@@ -288,7 +288,8 @@ export class Area extends PureComponent<Props, State> {
     } else if (isFunction(option)) {
       dotItem = option(props);
     } else {
-      dotItem = <Dot {...props} className="recharts-area-dot" />;
+      const className = clsx('recharts-area-dot', option ? (option as DotProps).className : '');
```
[the remark this answers is not in the document]
Comment (108yen) [reply src/cartesian/Area.tsx:291]: @ckifer 
Thank you for reviewing!
How about this?
```tsx
      const className = clsx('recharts-area-dot', typeof option !== 'boolean' ? option.className : '');
```

Should I fix line?
`Line.tsx`
```diff tsx
-      const className = clsx('recharts-line-dot', option ? (option as DotProps).className : '');
+      const className = clsx('recharts-line-dot',  typeof option !== 'boolean' ? option.className : '');
      dotItem = <Dot {...props} className={className} />;
```

[the remark this answers is not in the document]
Comment (108yen) [reply src/cartesian/Area.tsx]: @ckifer 
Fixed!

**Review by ckifer — APPROVED** [review]
nice, thank you!

**Pull request merged**